src/AppBundle/Controller/MainController.php

```diff
@@ -0,0 +1,210 @@
+        }
+
+        return new JsonResponse($todosArray);
+    }
+
+    /**
+     * @Route("/main/remove", name="remove_todo", options={"expose"=true})
+     */
+    public function removeAction(Request $request)
+    {   
+        $user = $this->getUser();
+        $em = $this->getDoctrine()->getManager();
+
+        $todoId = $request->request->get('id');
+        $todo = $em->getRepository('AppBundle:Todo')->find($todoId);
+
+        if (!$todo) {
+            throw $this->createNotFoundException(
+                'No product found for id '.$todoId
+            );
+        }
+
+        $user->removeTodo($todo);
+        $em->remove($todo);
+        $em->flush();
+
+
+        return new JsonResponse("Item removed Correctly");
+    }
+
+    /**
+     * @Route("/main/complete", name="complete_todo", options={"expose"=true})
+     */
+    public function completeAction(Request $request)
+    {   
+        $user = $this->getUser();
+        $em = $this->getDoctrine()->getManager();
+
+        $todoId = $request->request->get('id');
+        $todo = $em->getRepository('AppBundle:Todo')->find($todoId);
+
+        if (!$todo) {
+            throw $this->createNotFoundException(
+                'No product found for id '.$todoId
+            );
+        }
+
+        $todo->setCompleted(! $todo->getCompleted());
+        $em->flush();
+
+
+        return new JsonResponse("Completed state updated with".$todo->getCompleted());
+    }
+
+
+    /**
+     * @Route("/main/completeAll", name="complete_all", options={"expose"=true})
+     */
+    public function completeAllAction()
+    {
+
+        $em = $this->getDoctrine()->getManager();
+
+        $user = $this->getUser();
+        $todos = $user->getTodos();
+        $todos = $todos->toArray();
+
+        $actionDone = '';
+        $falseItemsNumber = 0;
+        foreach ($todos as $todo) {
+            if (! $todo->getCompleted()){
+                $falseItemsNumber++;
+            }
+        }
+
+        if ($falseItemsNumber == 0){
+
+            foreach ($todos as $todo) {
+                $todo->setCompleted(false);
+            }
+            $actionDone = 'makeAllFalse';
+
+        } else {
+
+            foreach ($todos as $todo) {
+                $todo->setCompleted(true);
+            }
+            $actionDone = 'makeAllTrue';
+
+        }
+
+        $em->flush();
+
+        return new JsonResponse(['action'=>$actionDone]);
+    }
+
+    /**
+     * @Route("/main/clearCompleted", name="clear_completed", options={"expose"=true})
+     */
+    public function clearCompletedAction()
+    {
+
+        $em = $this->getDoctrine()->getManager();
+
+        $user = $this->getUser();
+        $todos = $user->getTodos();
+        $todos = $todos->toArray();
+
+        $IdArray = [];
+        foreach ($todos as $todo) {
+            if ($todo->getCompleted()){
+                $IdArray[] = ['id' => $todo->getId()];
+
+                $user->removeTodo($todo);
```

Comment:
I wanted to keep $user->removeTodo so that i won't have any 'null' todos. Is that a bad idea? This also influenced my decision in the previous section when triggering exceptions